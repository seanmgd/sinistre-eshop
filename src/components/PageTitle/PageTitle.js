import styled from 'styled-components'

export const PageTitle = styled.h1`
  font-weight: 300;
  font-size: 28px;
  color: ${({ theme }) => theme.colors.black.base};
`
