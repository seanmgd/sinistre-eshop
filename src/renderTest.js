import { render as renderTest } from '@testing-library/react'
import { AppProviders } from './AppProviders'
import theme from './styles/theme'

// Export it first to allow overrides.
export * from '@testing-library/react'
export { renderInAppProviders as render }

export function renderInAppProviders(node, options) {
  const { rerender, ...res } = renderTest(
    <AppProviders theme={theme} children={node} />,
    options,
  )

  return {
    ...res,
    rerender: node => rerender(<AppProviders theme={theme} children={node} />),
  }
}
