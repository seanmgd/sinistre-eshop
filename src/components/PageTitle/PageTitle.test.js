import React from 'react'
import { render, fireEvent } from '../../renderTest'
import { PageTitle } from './PageTitle'

function createProps(props) {
  return { children: 'this is a test', ...props }
}

describe('<Avatar />', () => {
  it('should render a page title', () => {
    const props = createProps()
    const { container } = render(<PageTitle {...props} />)

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should spread other props on root element', () => {
    const props = createProps({
      style: { backgroundColor: 'red' },
      className: 'test-class',
      onClick: jest.fn(),
    })
    const { container, getByText } = render(<PageTitle {...props} />)

    expect(container.firstChild).toMatchSnapshot()

    fireEvent.click(getByText(props.children))
    expect(props.onClick).toHaveBeenCalledTimes(1)
  })
})
