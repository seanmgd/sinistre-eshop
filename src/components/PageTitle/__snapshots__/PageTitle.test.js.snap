// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageTitle /> should render a page title 1`] = `
<h1
  class="sc-bdfBwQ ctABtM"
>
  this is a test
</h1>
`;

exports[`<PageTitle /> should spread other props on root element 1`] = `
<h1
  class="sc-bdfBwQ ctABtM test-class"
  style="background-color: red;"
>
  this is a test
</h1>
`;
